tools/mo/openvino/tools/mo/main.py

```diff
@@ -321,6 +323,47 @@ def load_extensions(argv: argparse.Namespace, is_tf: bool, is_caffe: bool, is_mx
 def check_fallback(argv : argparse.Namespace):
     fallback_reasons = {}
 
+    def legacy_extensions_used(argv):
+        any_extension_used = hasattr(argv, 'extensions') and argv.extensions is not None and len(argv.extensions) > 0 \
+            and argv.extensions != import_extensions.default_path() # extensions arg has default value
+        if not any_extension_used:
+            return False
+        path = Path(argv.extensions)
+        new_ext_used = path.is_file() and (path.suffix == '.so' or path.suffix == '.dll')
+        if new_ext_used:
+            return False
+        return True
+
+    def is_new_json_config(json_file_path: Path):
```

Comment:
Looks like 
in case of new_json_config we need to use `moc_front_end .add_extension(JsonConfigExtension(json_file))`  in `prepare_ir `function below
in case of the new --extension option we need to use `moc_front_end.add_extension("path_to_lib) `in `prepare_ir `function below

Reply:
done